File: source/clibCore/Models/Entity/pageContentModel.vb

Option Explicit On
Option Strict On

Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Contensive.BaseClasses
Imports Contensive.Core.Controllers
Imports Newtonsoft.Json

Namespace Contensive.Core.Models.Entity
    Public Class pageContentModel
        Inherits baseModel
        '
        '====================================================================================================
        '-- const
        Public Const contentName As String = "page content"
        Public Const contentTableName As String = "ccpagecontent"
        Private Shadows Const contentDataSource As String = "default"
        '
        '====================================================================================================
        ' -- instance properties
        Public Property AllowBrief As Boolean
        Public Property AllowChildListDisplay As Boolean
        Public Property AllowEmailPage As Boolean
        Public Property AllowFeedback As Boolean
        Public Property AllowHitNotification As Boolean
        Public Property AllowInChildLists As Boolean
        Public Property AllowInMenus As Boolean
        Public Property AllowLastModifiedFooter As Boolean
        Public Property AllowMessageFooter As Boolean
        Public Property AllowMetaContentNoFollow As Boolean
        Public Property AllowMoreInfo As Boolean
        Public Property AllowPrinterVersion As Boolean
        Public Property AllowReturnLinkDisplay As Boolean
        Public Property AllowReviewedFooter As Boolean
        Public Property AllowSeeAlso As Boolean
        Public Property ArchiveParentID As Integer
        Public Property BlockContent As Boolean
        Public Property BlockPage As Boolean
        Public Property BlockSourceID As Integer
        Public Property BriefFilename As String
        Public Property ChildListInstanceOptions As String
        Public Property ChildListSortMethodID As Integer
        Public Property ChildPagesFound As Boolean
        Public Property Clicks As Integer
        Public Property ContactMemberID As Integer
        Public Property ContentPadding As Integer
        Public Property Copyfilename As New fieldTypeHTMLFile
        Public Property CustomBlockMessage As String
        Public Property DateArchive As Date
        Public Property DateExpires As Date
        Public Property DateReviewed As Date
        Public Property EditArchive As Boolean
        Public Property EditBlank As Boolean
        Public Property EditSourceID As Integer
        Public Property Headline As String
        Public Property imageFilename As String
        Public Property IsSecure As Boolean
        Public Property JSEndBody As String
        Public Property JSFilename As String
        Public Property JSHead As String
        Public Property JSOnLoad As String
        Public Property LinkAlias As String
        Public Property MenuHeadline As String
        Public Property metaDescription As String
        Public Property MetaKeywordList As String
        Public Property OtherHeadTags As String
        Public Property PageLink As String
        Public Property pageTitle As String
        Public Property ParentID As Integer
        Public Property ParentListName As String
        Public Property PubDate As Date
        Public Property RegistrationGroupID As Integer
        Public Property ReviewedBy As Integer
        Public Property TemplateID As Integer
        Public Property TriggerAddGroupID As Integer
        Public Property TriggerConditionGroupID As Integer
        Public Property TriggerConditionID As Integer
        Public Property TriggerRemoveGroupID As Integer
        Public Property TriggerSendSystemEmailID As Integer
        Public Property Viewings As Integer
        '
        '====================================================================================================
        Public Overloads Shared Function add(cpCore As coreClass) As pageContentModel
            Return add(Of pageContentModel)(cpCore)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function add(cpCore As coreClass, ByRef callersCacheNameList As List(Of String)) As pageContentModel
            Return add(Of pageContentModel)(cpCore, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cpCore As coreClass, recordId As Integer) As pageContentModel
            Return create(Of pageContentModel)(cpCore, recordId)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cpCore As coreClass, recordId As Integer, ByRef callersCacheNameList As List(Of String)) As pageContentModel
            Return create(Of pageContentModel)(cpCore, recordId, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cpCore As coreClass, recordGuid As String) As pageContentModel
            Return create(Of pageContentModel)(cpCore, recordGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cpCore As coreClass, recordGuid As String, ByRef callersCacheNameList As List(Of String)) As pageContentModel
            Return create(Of pageContentModel)(cpCore, recordGuid, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createByName(cpCore As coreClass, recordName As String) As pageContentModel
            Return createByName(Of pageContentModel)(cpCore, recordName)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createByName(cpCore As coreClass, recordName As String, ByRef callersCacheNameList As List(Of String)) As pageContentModel
            Return createByName(Of pageContentModel)(cpCore, recordName, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Sub save(cpCore As coreClass)
            MyBase.save(cpCore)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Sub delete(cpCore As coreClass, recordId As Integer)
            delete(Of pageContentModel)(cpCore, recordId)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Sub delete(cpCore As coreClass, ccGuid As String)
            delete(Of pageContentModel)(cpCore, ccGuid)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Function createList(cpCore As coreClass, sqlCriteria As String, sqlOrderBy As String, callersCacheNameList As List(Of String)) As List(Of pageContentModel)
            Return createList(Of pageContentModel)(cpCore, sqlCriteria, sqlOrderBy, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createList(cpCore As coreClass, sqlCriteria As String, sqlOrderBy As String) As List(Of pageContentModel)
            Return createList(Of pageContentModel)(cpCore, sqlCriteria, sqlOrderBy)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createList(cpCore As coreClass, sqlCriteria As String) As List(Of pageContentModel)
            Return createList(Of pageContentModel)(cpCore, sqlCriteria)
        End Function
        '
        '====================================================================================================
        Public Overloads Sub invalidatePrimaryCache(cpCore As coreClass, recordId As Integer)
            invalidateCacheSingleRecord(Of pageContentModel)(cpCore, recordId)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordName(cpcore As coreClass, recordId As Integer) As String
            Return baseModel.getRecordName(Of pageContentModel)(cpcore, recordId)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordName(cpcore As coreClass, ccGuid As String) As String
            Return baseModel.getRecordName(Of pageContentModel)(cpcore, ccGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordId(cpcore As coreClass, ccGuid As String) As Integer
            Return baseModel.getRecordId(Of pageContentModel)(cpcore, ccGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createDefault(cpcore As coreClass) As pageContentModel
            Return createDefault(Of pageContentModel)(cpcore)
        End Function
    End Class
End Namespace

File: source/core/Models/DbModels/pageContentModel.cs

using System;
using System.Reflection;
using System.Xml;
using System.Diagnostics;
using System.Linq;
using System.Text.RegularExpressions;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using Contensive.Processor;
using Contensive.Processor.Models.DbModels;
using Contensive.Processor.Controllers;
using static Contensive.Processor.Controllers.genericController;
using static Contensive.Processor.constants;
//
namespace Contensive.Processor.Models.DbModels {
    public class pageContentModel : baseModel {
        //
        //====================================================================================================
        //-- const
        public const string contentName = "page content";
        public const string contentTableName = "ccpagecontent";
        private const string contentDataSource = "default";
        //
        //====================================================================================================
        // -- instance properties
        public bool AllowBrief { get; set; }
        public bool AllowChildListDisplay { get; set; }
        public bool AllowFeedback { get; set; }
        public bool AllowHitNotification { get; set; }
        public bool AllowInChildLists { get; set; }
        public bool AllowInMenus { get; set; }
        public bool AllowLastModifiedFooter { get; set; }
        public bool AllowMessageFooter { get; set; }
        public bool AllowMetaContentNoFollow { get; set; }
        public bool AllowMoreInfo { get; set; }
        //// deprecated 20180701
        //public bool AllowPrinterVersion { get; set; }
        public bool AllowReturnLinkDisplay { get; set; }
        public bool AllowReviewedFooter { get; set; }
        public bool AllowSeeAlso { get; set; }
        public int ArchiveParentID { get; set; }
        public bool BlockContent { get; set; }
        public bool BlockPage { get; set; }
        public int BlockSourceID { get; set; }
        public string BriefFilename { get; set; }
        public string ChildListInstanceOptions { get; set; }
        public int ChildListSortMethodID { get; set; }
        public bool ChildPagesFound { get; set; }
        public int Clicks { get; set; }
        public int ContactMemberID { get; set; }
        public int ContentPadding { get; set; }
        public fieldTypeHTMLFile Copyfilename { get; set; } = new fieldTypeHTMLFile();
        public string CustomBlockMessage { get; set; }
        public DateTime DateArchive { get; set; }
        public DateTime DateExpires { get; set; }
        public DateTime DateReviewed { get; set; }



        public string Headline { get; set; }
        public string imageFilename { get; set; }
        public bool IsSecure { get; set; }
        public string JSEndBody { get; set; }
        public string JSFilename { get; set; }
        public string JSHead { get; set; }
        public string JSOnLoad { get; set; }
        public string LinkAlias { get; set; }
        public string MenuHeadline { get; set; }
        public string metaDescription { get; set; }
        public string MetaKeywordList { get; set; }
        public string OtherHeadTags { get; set; }
        public string PageLink { get; set; }
        public string pageTitle { get; set; }
        public int ParentID { get; set; }
        public string ParentListName { get; set; }
        public DateTime PubDate { get; set; }
        public int RegistrationGroupID { get; set; }
        public int ReviewedBy { get; set; }
        public int TemplateID { get; set; }
        public int TriggerAddGroupID { get; set; }
        public int TriggerConditionGroupID { get; set; }
        public int TriggerConditionID { get; set; }
        public int TriggerRemoveGroupID { get; set; }
        public int TriggerSendSystemEmailID { get; set; }
        public int Viewings { get; set; }
        //
        //====================================================================================================
        public static pageContentModel add(CoreController core) {
            return add<pageContentModel>(core);
        }
        //
        //====================================================================================================
        public static pageContentModel add(CoreController core, ref List<string> callersCacheNameList) {
            return add<pageContentModel>(core, ref callersCacheNameList);
        }
        //
        //====================================================================================================
        public static pageContentModel create(CoreController core, int recordId) {
            return create<pageContentModel>(core, recordId);
        }
        //
        //====================================================================================================
        public static pageContentModel create(CoreController core, int recordId, ref List<string> callersCacheNameList) {
            return create<pageContentModel>(core, recordId, ref callersCacheNameList);
        }
        //
        //====================================================================================================
        public static pageContentModel create(CoreController core, string recordGuid) {
            return create<pageContentModel>(core, recordGuid);
        }
        //
        //====================================================================================================
        public static pageContentModel create(CoreController core, string recordGuid, ref List<string> callersCacheNameList) {
            return create<pageContentModel>(core, recordGuid, ref callersCacheNameList);
        }
        //
        //====================================================================================================
        public static pageContentModel createByName(CoreController core, string recordName) {
            return createByName<pageContentModel>(core, recordName);
        }
        //
        //====================================================================================================
        public static pageContentModel createByName(CoreController core, string recordName, ref List<string> callersCacheNameList) {
            return createByName<pageContentModel>(core, recordName, ref callersCacheNameList);
        }
        //
        //====================================================================================================
        public new void save(CoreController core) {
            base.save(core);
        }
        //
        //====================================================================================================
        public static void delete(CoreController core, int recordId) {
            delete<pageContentModel>(core, recordId);
        }
        //
        //====================================================================================================
        public static void delete(CoreController core, string ccGuid) {
            delete<pageContentModel>(core, ccGuid);
        }
        //
        //====================================================================================================
        public static List<pageContentModel> createList(CoreController core, string sqlCriteria, string sqlOrderBy, List<string> callersCacheNameList) {
            return createList<pageContentModel>(core, sqlCriteria, sqlOrderBy, callersCacheNameList);
        }
        //
        //====================================================================================================
        public static List<pageContentModel> createList(CoreController core, string sqlCriteria, string sqlOrderBy) {
            return createList<pageContentModel>(core, sqlCriteria, sqlOrderBy);
        }
        //
        //====================================================================================================
        public static List<pageContentModel> createList(CoreController core, string sqlCriteria) {
            return createList<pageContentModel>(core, sqlCriteria);
        }
        //
        //====================================================================================================
        public void invalidatePrimaryCache(CoreController core, int recordId) {
            invalidateCacheSingleRecord<pageContentModel>(core, recordId);
        }
        //
        //====================================================================================================
        public static string getRecordName(CoreController core, int recordId) {
            return baseModel.getRecordName<pageContentModel>(core, recordId);
        }
        //
        //====================================================================================================
        public static string getRecordName(CoreController core, string ccGuid) {
            return baseModel.getRecordName<pageContentModel>(core, ccGuid);
        }
        //
        //====================================================================================================
        public static int getRecordId(CoreController core, string ccGuid) {
            return baseModel.getRecordId<pageContentModel>(core, ccGuid);
        }
        //
        //====================================================================================================
        public static pageContentModel createDefault(CoreController core) {
            return createDefault<pageContentModel>(core);
        }
    }
}
